rust/lance-linalg/src/simd/f16.c

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+
+#ifndef LANES
+#define LANES 8
+#endif
+
+/// Works on NEON + FP16 or AVX512FP16
+//
+// Please make sure run "cargo bench --bench norm_l2" on both Apple Silicon and
+// X86_64, before you change this function.
+_Float16 norm_l2_f16(_Float16* data, size_t dimension) {
+  _Float16 sums[LANES];
+#pragma clang loop unroll(enable) vectorize(enable)
+  for (size_t i = 0; i < LANES; i++) {
```

Comment:
Style nit: don't use brackets for single line loops.

Reply:
fixed. Is it a C special thing? C++ seems encourage to sue brackets nonetheless

Reply:
Different style guide will have different recommendations on this one. E.g. the LLVM style guide says no braces for single line; google says yes.

I just think it's a little cleaner.

Reply:
The full guide is here: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements